### PR TITLE
[backend] send encoded command into implant

### LINKED ISSUE #1831: Raise the interpolation and obfuscation of commands to the OpenBAS platform level
## Description

Now
- OpenBAS put it the job table with the command in plain text
- The signature mechanics, retrieve this command and encode it
- The implant retrieves this command, interpolates the arguments and encodes the command

Later
- OpenBAS feeds the job table with the interpolation and encoding of the command.
- The signature mechanics recover the command clearly and the command reworked.
- The implant retrieves the command directly reworked by OpenBAS.

### PR DESCRIPTION
**Pull request by MarineLeM**
### Proposed changes

Improvements :
* Move inject service file under rest>inject>service
* Remove useless table injects_payloads ( legacy )
* Remove hardcoded payload type comparisons (replace with enum values).

Code related to the issue :
* Add new endpoint `@GetMapping(INJECT_URI + "/{injectId}/executable-payload")` 
used to get payload with encoded command

### Related issues

* Closes #1831 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion) => in progress
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

⚠️ need to be merge with https://github.com/OpenBAS-Platform/implant/pull/22

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java [hunk 1]
```diff
@@ -170,6 +172,13 @@ public Inject tryInject(@PathVariable String injectId) {
     return atomicTestingService.tryInject(injectId);
   }
 
+  @GetMapping(INJECT_URI + "/{injectId}/executable-payload")
```
Comment (Dimfacion) [inline openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java:175]: Could you add a Tracing annotation (you can have a look at what's already in ExerciseApi.java for an example) ?

Comment (Dimfacion) [reply openbas-api/src/main/java/io/openbas/rest/inject/InjectApi.java:175]: Also, if you want to test, if your dev environment's docker-compose is up to date, you can reach localhost:16686. I'll do a demo whenever it's available for prod envs.

File: openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ExecutableInjectService {
+
+  private final InjectRepository injectRepository;
+
+  private List<String> getArgumentsFromCommandLines(String command) {
+    Pattern pattern = Pattern.compile("#\\{([^#{}]+)}");
```
Comment (Dimfacion) [inline openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java:22]: Pattern compilation is cpu consuming (https://rules.sonarsource.com/java/tag/performance/RSPEC-4248/) You should pre compile it if possible.

File: openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java [hunk 3]
```diff
@@ -145,15 +145,15 @@ public ExecutionProcess process(
                             List<InjectExpectationSignature> injectExpectationSignatures =
                                 new ArrayList<>();
                             if (injectorContract.getPayload() != null) {
-                              switch (injectorContract.getPayload().getType()) {
-                                case "Command":
+                              switch (injectorContract.getPayload().getTypeEnum()) {
```
Comment (Dimfacion) [inline openbas-api/src/main/java/io/openbas/injectors/caldera/CalderaExecutor.java:148]: **praise :** Good point

**Comment by MarineLeM** [comment]
@Dimfacion I took into account your observations. Please let me know if you notice anything else. Thank you!🙏

File: openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  private String processAndEncodeCommand(
+      String command,
+      String executor,
+      List<PayloadArgument> defaultArguments,
+      ObjectNode injectContent,
+      String obfuscator) {
+    String computedCommand = replaceArgumentsByValue(command, defaultArguments, injectContent);
+
+    if (executor.equals("cmd")) {
+      computedCommand = computedCommand.replace("\n", " & ");
```
Comment (damgouj) [inline openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java:69]: I think maybe you forgot the .trim() (it was in the implant code for cmd command)

**Review by damgouj — CHANGES_REQUESTED** [review]

File: openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java (hunk 4 above)
Comment (MarineLeM) [reply openbas-api/src/main/java/io/openbas/rest/inject/service/ExecutableInjectService.java:69]: Thank you i didn't see that !

**Review by damgouj — APPROVED** [review]

**Review by Dimfacion — APPROVED** [review]

**Pull request merged**